build error: duplicate swap()

When building an application with the foolowing libs in  components.mk:
```ARDUINO_LIBRARIES := ArduinoJson6 Adafruit_SSD1306```

then the following code fails:
```
#include <SmingCore.h>
#include <Libraries/Adafruit_SSD1306/Adafruit_SSD1306.h>
#include <ArduinoJson.h>
```
with:
```
In file included from /home/martijn/espdev/Sming/Sming/Libraries/Adafruit_SSD1306/Adafruit_SSD1306.h:49:0,
                 from /home/martijn/espdev/projects3/bug01/app/application.cpp:3:
/home/martijn/espdev/Sming/Sming/Libraries/Adafruit_GFX/Adafruit_GFX.h:11:20: error: expected unqualified-id before '{' token
 #define swap(a, b) { int16_t t = a; a = b; b = t; }
                    ^
In file included from /home/martijn/espdev/Sming/Sming/Libraries/ArduinoJson6/ArduinoJson/src/ArduinoJson/MsgPack/endianess.hpp:8:0,
                 from /home/martijn/espdev/Sming/Sming/Libraries/ArduinoJson6/include/../ArduinoJson/src/ArduinoJson/MsgPack/MsgPackDeserializer.hpp:11,
                 from /home/martijn/espdev/Sming/Sming/Libraries/ArduinoJson6/include/../ArduinoJson/src/ArduinoJson.hpp:36,
                 from /home/martijn/espdev/Sming/Sming/Libraries/ArduinoJson6/include/../ArduinoJson/src/ArduinoJson.h:9,
                 from /home/martijn/espdev/Sming/Sming/Libraries/ArduinoJson6/include/ArduinoJson.h:38,
                 from /home/martijn/espdev/projects3/bug01/app/application.cpp:4:
/home/martijn/espdev/Sming/Sming/Libraries/ArduinoJson6/ArduinoJson/src/ArduinoJson/Polyfills/utility.hpp:9:30: error: expected unqualified-id before '{' token
 inline void swap(T& a, T& b) {
                              ^
/home/martijn/espdev/Sming/Sming/component-wrapper.mk:155: recipe for target 'app/application.o' failed
```


It seems the swap() define in Adafruit_GFX.h is a (bad) duplicate.
Change the define into an inline function inside the Adafruit_GFX class fixes the problem.
```
 static inline void swap(int16_t &a, int16_t &b) {int16_t t = a; a = b; b = t;}
```

There may exist a better solution...

--- a/Sming/Libraries/Adafruit_GFX/Adafruit_GFX.h
+++ b/Sming/Libraries/Adafruit_GFX/Adafruit_GFX.h
@@ -8,8 +8,6 @@
  #include "WProgram.h"
 #endif
 
-#define swap(a, b) { int16_t t = a; a = b; b = t; }
-
 class Adafruit_GFX : public Print {
 
  public:

--- a/Sming/Libraries/ArduCAM/ArduCAM.h
+++ b/Sming/Libraries/ArduCAM/ArduCAM.h
@@ -101,8 +101,6 @@
 #define cport(port, data) port &= data
 #define sport(port, data) port |= data
 
-#define swap(type, i, j) {type t = i; i = j; j = t;}
-
 #define fontbyte(x) pgm_read_byte(&cfont.font[x])  
 
 #define regtype volatile uint8_t
@@ -119,8 +117,6 @@
 
 #define cport(port, data) port &= data
 #define sport(port, data) port |= data
-
-#define swap(type, i, j) {type t = i; i = j; j = t;}
 
 #define fontbyte(x) cfont.font[x]  
 
@@ -156,8 +152,6 @@
 
 #define cport(port, data) port &= data
 #define sport(port, data) port |= data
-
-#define swap(type, i, j) {type t = i; i = j; j = t;}
 
 #define fontbyte(x) cfont.font[x]
 

--- a/Sming/Wiring/WiringFrameworkDependencies.h
+++ b/Sming/Wiring/WiringFrameworkDependencies.h
@@ -26,3 +26,4 @@ using std::isinf;
 using std::isnan;
 using std::max;
 using std::min;
+using std::swap;

